### ISSUE TITLE
Google searches still lead to 0.18 docs

### ISSUE THREAD

**Issue by bplommer** [http4s/http4s#5715]
For example, the top goole result for "http4s cors" (and the only one leading to http4s docs) is to https://http4s.org/v0.18/cors/. Fixing this / stopping it from happening will need some SEO expertise unfortunately (I've seen similar things for Doobie).

**Comment by bplommer** [comment]
Perhaps the best solution to this is to take down the 0.18 docs and have the URLs redirect to a later version?

**Comment by armanbilge** [comment]
https://developers.google.com/search/docs/advanced/crawling/block-indexing

Seems it's as simple as adding this tag to the relevant pages.
```html
<meta name="robots" content="noindex">
```

**Comment by bplommer** [comment]
Won't that result in no docs showing up though?

**Comment by armanbilge** [comment]
Well, I was thinking we'd just put it on the v0.18 docs (and any other old ones). Can't be any worse than taking them down?

**Comment by bplommer** [comment]
I was thinking that if we make the 0.18 urls redirect to newer versions then we'll keep the existing search result positions but they'll go somewhere more relevant. But maybe that's not how it works?

**Comment by armanbilge** [comment]
My search results look like this. First entry is 0.18, second is 1.0. So I assume that would replace it. But I won't claim to have any idea how these work :)

![Screen Shot 2022-01-02 at 11 03 23 AM](https://user-images.githubusercontent.com/3119428/147881622-64545296-e2cc-4162-818d-c401192de802.png)

**Comment by bplommer** [comment]
Ok yes, I think you're right 🙂

**Comment by rossabaker** [comment]
Some projects do a nice thing where the top of obsolete docs redirects to the latest with an ugly banner at the top.  We could implement that now, except GitHub pages doesn't support a 3xx we could point.  Rewrites are why we started hosting the site on Netlify, but bus factor of administration is an issue there, so we'd like to wean ourselves from that.  I don't know how to point 0.18 at the latest and _keep_ it there, without republishing from 0.18 every time there's a new version.  I guess it could always be N hops away.

Sadly, the recent changes to 0.18 came up because we have a service on that still, and a new developer doing their job found a mistake in it.  It would be a shame for it to go away entirely.

**Comment by armanbilge** [comment]
>  It would be a shame for it to go away entirely.

IMHO I think leaving them live but off Google results is reasonable compromise.

**Comment by bplommer** [comment]
>  I don't know how to point 0.18 at the latest and keep it there, without republishing from 0.18 every time there's a new version. I guess it could always be N hops away.

We already have a `/latest` path for the latest docs - just point to that?

**Comment by rossabaker** [comment]
> IMHO I think leaving them live but off Google results is reasonable compromise.

Agreed.  Does that robots work at a directory level, or does it need to be in the server root?

> We already have a /latest path for the latest docs - just point to that?

That feature requires Netlify, unless I missed some new development.

**Comment by armanbilge** [comment]
My understanding of their documentation is that we need to include the meta-tag on each affected page.
https://developers.google.com/search/docs/advanced/crawling/block-indexing
```html
<meta name="robots" content="noindex">
```

**Comment by rossabaker** [comment]
We should be able to do that easily enough from one of the Hugo templates.

**Comment by armanbilge** [comment]
I'm [reading](https://yoast.com/rel-canonical/) about `<link rel="canonical" href="...">` and it's an alternative we may want to consider to `<meta name="robots" content="noindex">`.

IIUC when you have more than one version of the same page, linking "canonical" version of that page indicates to search engines which one they should show in search results.

> What this does is “merge” the two pages into one from a search engine’s perspective. It’s a “soft redirect” without actually redirecting the user. Links to both URLs now count as the single, canonical version of the URL.

Of course this is a matter of opinion whether e.g the v0.23 site is the canonical for all the http4s versions. But to me this seems like a reasonable choice: it will condense all of our search engine results that may currently be scattered across versions, and viewers can use the version picker.

Obviously this would be something to upstream into sbt-typelevel or probably Laika.

**Comment by rossabaker** [comment]
It looks like maybe Read The Docs uses this approach: https://haha.readthedocs.io/en/latest/canonical.html
